### PR TITLE
Add SIRF/CIL integration tests

### PR DESCRIPTION
**Pull request by paskino**
This reinstates the tests for the SIRF/CIL integration present in CIL.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
   - siemens_to_ismrmrd: 8bb8b08f53ce73c2de9ba5f47f1532f96292d92b
   - ISMRMRD: v1.13.2
   - SIRF: v3.5.0
+  - CIL: e7e137b97d403576502ffc768e596ec1f66e038a
```
Comment (paskino) [inline CHANGES.md:25]:
```suggestion
  - CIL: 0ba2f8e2935db66ec3dfe8c88e407e8d0eb5609f
```

File: version_config.cmake [hunk 2]
```diff
@@ -148,7 +148,7 @@ set(DEFAULT_JSON_TAG v3.10.4)
 # CCPi CIL
 # minimum supported version of CIL supported is > 22.1.0 or from commit a6062410028c9872c5b355be40b96ed1497fed2a
 set(DEFAULT_CIL_URL https://github.com/TomographicImaging/CIL.git)
-set(DEFAULT_CIL_TAG a6062410028c9872c5b355be40b96ed1497fed2a)
+set(DEFAULT_CIL_TAG test_SIRF_fix)
```
Comment (paskino) [inline version_config.cmake:151]:
```suggestion
set(DEFAULT_CIL_TAG 0ba2f8e2935db66ec3dfe8c88e407e8d0eb5609f)
```

**Comment by casperdcl** [comment]
fixes https://github.com/SyneRBI/SIRF-SuperBuild/pull/835#issuecomment-1709896466.

[heisenbug failure](https://github.com/SyneRBI/SIRF-SuperBuild/actions/runs/6118889932/job/16607770493) disappeared [after restarting](https://github.com/SyneRBI/SIRF-SuperBuild/actions/runs/6118889932/job/16611840907)

**Review by casperdcl — APPROVED** [review]

**Comment by KrisThielemans** [comment]
> fixes [#835 (comment)](https://github.com/SyneRBI/SIRF-SuperBuild/pull/835#issuecomment-1709896466).

It doesn't seem to really fixe that comment?  This reinstates the CIL tests if we build it via the SuperBuild (whcih is what we do on GHA and possibly the VM). But the comment was intended to problem when we install CIL via conda (which is what we do on docker)

**Review by KrisThielemans — APPROVED** [review]

**Comment by casperdcl** [comment]
sure I suppose it's more related to #836 & #837

**Pull request merged**